src/state/opamSysInteract.ml

```diff
@@ -129,6 +130,7 @@ let family ~env () =
       failwith
         "External dependency handling for macOS requires either \
          MacPorts or Homebrew - neither could be found"
+    | "nixos" -> Nixos
```

Comment:
This does lead to the question of what opam will/should do if running on macOS with nix-darwin (as I do).  I have both Homebrew and nix-darwin installed -- it would be nice to be able to explicitly configure which depext backend to use rather than have it inferred by opam.

Reply:
Regarding what it will do, I believe `OpamSysPoll.os_family env` will return [`homebrew`/`macports`](https://github.com/ocaml/opam/blob/3434460674f4df689dc982971ba9c2e96d349902/src/state/opamSysPoll.ml#L94-L97), so the behavior will be unchanged by this PR.

Regarding what it should do, I don't think there's much to be gained from installing depexts on macOS through `nix-env`. My opinion is `nix-env` isn't the 'right' way to use Nix, but is a useful function to provide an interface similar to traditional package managers. As such, this PR is useful in that it provides the path of least resistance to getting a project building on a NixOS system with existing OCaml tooling, as opposed to using one of the various projects for building OCaml projects with Nix.

Regarding explicitly configuring which backend to use, opam is already using homebrew by default and macports if homebrew is unavailable. Using `nix-env` could be useful in providing a third option (with a very minor change), i.e.:
```
else if OpamSystem.resolve_command "nix-env" <> None then Some "nix"
```
Perhaps explicitly configuring which depext backend to use could be another PR.

Reply:
Thanks Ryan, that all sounds very sensible to me.